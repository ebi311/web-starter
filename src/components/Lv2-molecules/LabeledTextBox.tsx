import React, { useCallback } from 'react';
import { styled } from '../Theme';
import TextBox from '../Lv1-atoms/TextBox';
import InputLabel from '../Lv1-atoms/InputLabel';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

interface IProps {
  className?: string;
  label: string;
  onChange: (value: string) => void;
  value: string;
}

const LabeledTetBox: React.FC<IProps> = props => {
  const onChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    props.onChange(e.currentTarget.value);
  }, []);
  const onBlur = useCallback((e: React.FocusEvent<HTMLInputElement>) => {
    props.onChange(e.currentTarget.value);
  }, []);
  return (
    <Container className={props.className}>
      <InputLabel>{props.label}</InputLabel>
      <TextBox value={props.value} onChange={onChange} onBlur={onBlur} />
    </Container>
  );
};

export default LabeledTetBox;
