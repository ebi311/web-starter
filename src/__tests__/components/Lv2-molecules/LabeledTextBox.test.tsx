import React from 'react';
import { shallow } from 'enzyme';
import LabeledTextBox from '../../../components/Lv2-molecules/LabeledTextBox';

describe('LabeledTextBox', () => {
  test('success', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <LabeledTextBox label="l001" value="v001" onChange={onChange} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
    const textbox = wrapper.find('TextBox');
    textbox.simulate('change', { currentTarget: { value: 't001' } });
    expect(onChange).toBeCalledWith('t001');
    onChange.mockClear();
    textbox.simulate('blur', { currentTarget: { value: 't002' } });
    expect(onChange).toBeCalledWith('t002');
  });
});
