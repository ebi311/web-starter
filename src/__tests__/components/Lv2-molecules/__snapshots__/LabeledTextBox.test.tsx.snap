// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledTextBox success 1`] = `
<div
  className="sc-AxiKw dIfukv"
>
  <InputLabel>
    l001
  </InputLabel>
  <TextBox
    onBlur={[Function]}
    onChange={[Function]}
    value="v001"
  />
</div>
`;
